File: Easy/121-best-time-to-buy-and-sell-stock/try-5.py
/*
 * Problem #121: Best Time to Buy and Sell Stock
 * Difficulty: Easy
 * Submission: Try 5
 * status: Accepted
 * Language: python3
 * Date: 02/12/2024, 18:43:52
 * Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 */

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        m=s=0
        for i in range(1,len(prices)):
            
            m=max(s:=max(s:=s+prices[i]-prices[i-1],0),m)
        return m

            

        

File: Easy/242-valid-anagram/try-5.cpp
/*
 * Problem #242: Valid Anagram
 * Difficulty: Easy
 * Submission: Try 5
 * status: Accepted
 * Language: cpp
 * Date: 22/11/2024, 21:33:50
 * Link: https://leetcode.com/problems/valid-anagram/
 */

class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
        unordered_map<char,int> map;
        for(int i=0;i<s.size();i++){
            map[s[i]]++;
            map[t[i]]--;
        }
        for(int i : t){
            if(map[i]!=0){
                return false;
            }
        }
        return true;
    }
        
};
